### DISCUSSION TITLE
In 16.9.1 p-dropdown label does not update with the recently selected value from the dropdown

### DISCUSSION THREAD

**Discussion by aabot99** [primefaces/community#657] [PrimeNG] [answered]
I just upgraded from 16.0.2 to 16.9.1 and the p-dropdown selected value is not displayed in the dropdown.

In this example, I clicked on the dropdown and select the "linux" entry:
![image](https://github.com/primefaces/community/assets/152433894/bc804ff9-5d32-44a3-9e1b-0995de7b093d)

The value does not display in the p-dropdown input field.  I clicked on the dropdown to show the correct value is highlighted:
![image](https://github.com/primefaces/community/assets/152433894/a6660122-2254-4bdc-96a0-317c6510c5be)

Here is the HTML snippet which is the same between the two different primeng revs.  The onChange handler just sets a flag to true.
![image](https://github.com/primefaces/community/assets/152433894/6209c459-9006-4ec3-844a-d1fb04f99c48)

UPDATE: I played with the different properties and finally discovered when you remove optionLabel and optionValue, the dropdown shows the selected value and updates after you set a new value.  My options are an array of {label,value} so this looks like a bug to me.  Will there be a patch update to fix this?

**Comment by cetincakiroglu** [comment]
Hi,

Could you please create a stackblitz example so we can identify the issue clearly?

**Comment by aabot99** [comment]
Sorry for the delay.  I created a stackblitz at https://stackblitz.com/edit/zkmdqg-wtdqca?file=src%2Fapp%2Fdemo%2Fdropdown-template-demo.html that shows the issue.  Trying a few more options, I narrowed down the issue to the optionLabel and optionValue properties.   When I remove those two properties from the p-dropdown element, the dropdown works as expected and I confirmed it works in my original project.  My mode of operation is the options are an array of objects and the model is the value attribute not the option object entry. 

In other places in my code, I have the model point to the options array object directly.

Are the optionLabel and optionValue properties being depreciated?

**Comment by cetincakiroglu** [comment, marked as answer]
Hi, 

This issue is fixed in v17, optionLabel and optionValue properties are not deprecated.

**Reply by aabot99** [reply]
Thanks.  I upgraded from 16.0.2 to the latest 17.3.1 (using freya template), just needed a few tweaks, and it is working now.

**Reply by stevenpotts** [reply]
for me, 17.5.0 is broken, but 17.3.1 works.
